llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -737,9 +737,11 @@ bool SPIRVInstructionSelector::spvSelect(Register ResVReg,
 
   case TargetOpcode::G_FMINNUM:
   case TargetOpcode::G_FMINIMUM:
+  case TargetOpcode::G_FMINIMUMNUM:
```

Comment:
Something isn't right on how this whole thing is handled:

If both operand are NaNs, behaviour should match
If one is NaN, `fminimumnum` and ocl returns the other operand but glsl can return either operand, the other 2 llvm intrinsic have different behaviour if it is sNaN or qNan.

Behaviour with -0 / 0 isn't set for ocl as well.

Reply:
> Something isn't right on how this whole thing is handled:

Yes, I think GMINNUM and G_FMINIMUM need fix.

> If one is NaN, `fminimumnum` and ocl returns the other operand but glsl can return either operand, the other 2 llvm intrinsic have different behaviour if it is sNaN or qNan.

thanks @Naghasan, I was not aware of GLSL behavior.